packages/upload/src/vaadin-upload.js

```diff
@@ -879,12 +879,13 @@ class Upload extends ElementMixin(ThemableMixin(ControllerMixin(PolymerElement))
       );
       return;
     }
-    const fileExt = file.name.match(/\.[^.]*$|$/u)[0];
     // Escape regex operators common to mime types
     const escapedAccept = this.accept.replace(/[+.]/gu, '\\$&');
```

Comment:
It probably makes sense to escape `.` to avoid misinterpretation.

```suggestion
    const escapedAccept = this.accept.replace(/[+\.]/gu, '\\$&');
```

Reply:
IIRC it results in an error as a redundant escape, but I will commit this and let's see.

Reply:
Oh, got it. We can leave it as is, I guess